src/AgentCommerce/SalesChannel/CheckoutRoute.php

```diff
@@ -20,11 +31,56 @@
  */
 #[Package('checkout')]
 #[Route(defaults: ['_routeScope' => ['paypal-agent'], '_agentScope' => [AgentSource::SCOPE_CART, AgentSource::SCOPE_CHECKOUT]])]
-class CheckoutRoute
+class CheckoutRoute extends AbstractAgentCommerceRoute
 {
+    public function __construct(
+        private readonly AbstractCartOrderRoute $orderRoute,
+        private readonly CartService $cartService,
+        private readonly OrderResource $orderResource,
+        private readonly PayPalPaymentHandler $paymentHandler,
+        private readonly PayPalCartTransformer $cartTransformer,
+    ) {
+    }
+
     #[Route('/api/paypal/v1/merchant-cart/{token}/checkout', name: 'api.paypal.merchant-cart.checkout', methods: [Request::METHOD_POST])]
-    public function checkout(string $token, Request $request, Context $context): AgentCartResponse
+    public function checkout(string $token, Request $request, SalesChannelContext $context): AgentCartResponse
     {
-        return new AgentCartResponse(new PayPalCart());
+        $extractedToken = CartTokenValidator::validateCartToken($token);
+
+        $cart = $this->cartService->getCart($extractedToken, $context);
+        if (!$cart->getLineItems()->count()) {
+            // We don't create a cart with empty items. So it must be created.
+            throw AgentException::cartNotFound($token);
+        }
+
+        $order = $this->orderRoute
+            ->order($cart, $context, new RequestDataBag($request->request->all()))
+            ->getOrder();
+
+        $primaryTransaction = $order->getPrimaryOrderTransactionId();
+
+        if (!$primaryTransaction) {
+            throw AgentException::orderSystemError();
+        }
+
+        $body = \json_decode($request->getContent(), true, flags: \JSON_THROW_ON_ERROR);
+
+        $payPalOrder = $this->orderResource->get($body['payment_method']['token'], $context->getSalesChannelId());
```

Comment:
shouldn't this work? Maybe you're missing the content-type header in your tests
```suggestion
        $payPalOrder = $this->orderResource->get($request->get('payment_method')['token'], $context->getSalesChannelId());
```

Reply:
$request->get is actually considered internal by Symfony

Reply:
ah yes sorry, I meant `$request->request->get` :thinking: